File: solutions/001.two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var ans = [];
    var hash = {};
    for (var i = 0; i < nums.length; i++) {
        if (typeof hash[nums[i]] !== "undefined") {
            ans[0] = hash[nums[i]];
            ans[1] = i;
        }
        else {
            hash[target - nums[i]] = i;
        }
    }
    return ans;
};

File: solutions/001.two-sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  ret = []
  hash = Hash.new
  nums.each_with_index do |val, idx|
    if hash.has_key?(val)
      ret[1] = idx+1
      ret[0] = hash[val]+1
      return ret
    else
      hash[target-val] = idx
    end
  end
end